Add basic test coverage for auth and tasks

MeghaMuralidharan opened the pull request:
- Add pytest test suite with httpx.AsyncClient and pytest-asyncio
- Test user registration endpoint
- Test login and token generation  
- Test task creation and retrieval
- Use test DB from .env as TEST_DB_URL
- Add proper test configuration with conftest.py
 PR for issue #10

MeghaMuralidharan committed:
Add basic test coverage for auth and tasks

iinava commented:
<img width="1853" height="931" alt="Screenshot 2025-11-21 131527" src="https://github.com/user-attachments/assets/45272938-bcd8-481e-b04d-477bec26d164" />

MeghaMuralidharan committed:
docs: Add changelog and document issue #10

iinava commented on tests/test_auth.py:
```diff
@@ -0,0 +1,61 @@
+import os
+
+# Test DB URL from environment as required by Issue 10
+TEST_DB_URL = os.getenv("TEST_DB_URL", "postgresql+asyncpg://admin:123456@localhost:5432/test_db")
```
TEST_DB_URL not applied
Read from env but not used anywhere to override the database
Tests likely use the default database from settings
need to override get_db or create new dependency

iinava commented on tests/conftest.py:
```diff
@@ -0,0 +1,15 @@
+from app.main import app
+
+@pytest.fixture
+def client():
```
Current: TestClient (synchronous) in conftest.py
Required: httpx.AsyncClient for async endpoints

use  httpx.AsyncClient as mentioned in issue

iinava commented on tests/test_task.py:
```diff
@@ -0,0 +1,31 @@
+    print("✅ Task retrieval requires authentication")
+    
+    # Test endpoints exist (not 404)
+    assert response.status_code != 404
```
Please test the actual task creation as well, not just the existence of the routes

What it should cover
Register/login a user to get a token
Create a task with the token
Retrieve tasks with the token
Verify the created task appears in the results

MeghaMuralidharan committed:
• Add user registration and login tests
• Add task creation and retrieval tests
• Apply TEST_DB_URL from environment
• Add complete workflow test
• All tests passing
